### PR TITLE
Update readmes and simplify repo

### PR DESCRIPTION
**Pull request by MattsonCam**
This pr focuses on updating the readmes in each module, except the `3.figures` module, in addition to simplifying the repo structure. I plan on including more analysis from my local pc in the `0.data_analysis` module on gh in future pr(s). Please provide any constructive suggestions you may have while reviewing this pr.

### CODE REVIEW

File: 0.data_analysis/README.md
Comment (jenna-tomkinson) [inline 0.data_analysis/README.md]: Recommend instead of saying that this is "not included" in the other repo, it might be more helpful to link to the `4.analyze_data` module and state what analysis was perform that is different from this repo (e.g., UMAPs, linear models, etc.).

We might also consider renaming one of these folders in one repo, either this one or the other one. This doesn't have to be done in this PR, but it will be good to discuss once we are ready to preprint the manuscript.

File: 0.data_analysis/README.md
Comment (jenna-tomkinson) [inline 0.data_analysis/README.md]:
```suggestion
We organize this module into two correlation analyses:
```

File: 0.data_analysis/README.md
Comment (jenna-tomkinson) [inline 0.data_analysis/README.md]:
```suggestion
1. `plate_5_characteristics` explores similarities between genotype groups using cell morphologies aggregated to the well-level, post-feature selection.
```

File: 1.train_models/README.md
Comment (jenna-tomkinson) [inline 1.train_models/README.md]:
```suggestion
After substantial development, we have developed and optimized our final NF1 model with the same architecture as a logistic regression.
```
Quick question on this statement:
> "same architecture as a logistic regression"

When you say this, it sounds like we made a whole new ML model. Remind me, do you use the sklearn package to create a logistic regression or was there another method?

File: 1.train_models/README.md
Comment (jenna-tomkinson) [inline 1.train_models/README.md]:
```suggestion
The training and testing sets were choosen to create a 90-10 percent train-test split between wells, while approximately retaining a uniform genotype class distribution of cells within each data split.
```

Comment (jenna-tomkinson) [reply 1.train_models/README.md]: Recommend staying consistent with past or present tense. Also might not need to include percent, so feel free to add or ignore.

File: 1.train_models/README.md
Comment (jenna-tomkinson) [inline 1.train_models/README.md]:
```suggestion
After optimization, we retrain the NF1 model with the optimal regularization strength using the entire dataset, but excluding the test set.
```

File: 1.train_models/README.md
Comment (jenna-tomkinson) [inline 1.train_models/README.md]:
```suggestion
During optimization of the NF1 model, we save the predicted genotype probabilities, true genotype labels, and the predicted genotype labels of the training, validation, and testing datasets for later evaluation.
```

File: 2.evaluate_model/README.md
Comment (jenna-tomkinson) [inline 2.evaluate_model/README.md]:
```suggestion
After training the NF1 model described in [1.train_models]("../1.train_models), and saving the results, we evaluate the performance of the NF1 model.
```

File: 2.evaluate_model/README.md
Comment (jenna-tomkinson) [inline 2.evaluate_model/README.md]:
```suggestion
In addition to these changes, we save the feature importances of the model to gather insights about key morphology differences.
```

File: README.md
Comment (jenna-tomkinson) [inline README.md]: Recommend including a short description of this repo in between these two sections. I think you include it after the `Repository Structure` title, so it might be better moved above it.

File: README.md
Comment (jenna-tomkinson) [inline README.md]:
```suggestion
Here we analyze NF1 morphology data from the [NF1 Cell painting Data](https://github.com/WayScience/nf1_cellpainting_data) repo. 
Our analysis is categorized as follows:
```
One sentence per line.

File: README.md
Comment (jenna-tomkinson) [inline README.md]:
```suggestion
| [3.figures](./3.figures/) | Generate manuscript figures | We interpret the results from the model by generating figures to include in the manuscript. |
```

**Review by jenna-tomkinson — APPROVED** [review]
I am approving this PR since I believe the comments I made don't need to be revisited for another review. I added several comments on the READMEs. Overall, great job on keeping them concise! My one main overall comment is that I would recommend including in each of the module READMEs a section on how to reproduce the code in that folder (e.g., a bash script).

Once you address these comments, feel free to merge! Nice job! :tada:

File: 0.data_analysis/README.md
Comment (MattsonCam) [reply 0.data_analysis/README.md]: > Recommend instead of saying that this is "not included" in the other repo, it might be more helpful to link to the `4.analyze_data` module and state what analysis was perform that is different from this repo (e.g., UMAPs, linear models, etc.).

Maybe we could list some examples, and then refer these readers to both modules READMEs' for more information by providing links? I think this would help keep the READMEs more DRY.
> We might also consider renaming one of these folders in one repo, either this one or the other one. This doesn't have to be done in this PR, but it will be good to discuss once we are ready to preprint the manuscript.

Sounds good, we can do that. I think we will also want to rename the analysis repo

File: 1.train_models/README.md
Comment (MattsonCam) [reply 1.train_models/README.md]: This is just another way of saying I used a logistic regression in a more formal way. Also, it is an sklearn model

File: 1.train_models/README.md
Comment (MattsonCam) [reply 1.train_models/README.md]: I was trying to create an active voice, but your suggestion looks better

File: README.md
Comment (MattsonCam) [reply README.md]: Should this module be "advertised" as containing only manuscript figures, or more generally, those results we want to interpret from the model? It may be fine to "advertise" this module as intended for model manuscript figures, since the manuscript will focus on interpreting the results of the model

File: README.md
Comment (MattsonCam) [reply README.md]: I restructured this a little. Lmk what you think!

**Comment by MattsonCam** [comment]
Hi @jenna-tomkinson, just requested a re-review for the open conversations. If you want to add a README for `3.figures` in this pr, feel free as well!

File: 0.data_analysis/README.md
Comment (jenna-tomkinson) [reply 0.data_analysis/README.md]: > Maybe we could list some examples, and then refer these readers to both modules READMEs' for more information by providing links? I think this would help keep the READMEs more DRY.

I can see that. I recommend not mentioning the image profiling repo at all here then and just focus on discussing what is going on in that module.

File: 0.data_analysis/README.md
Comment (jenna-tomkinson) [inline 0.data_analysis/README.md]: Recommend updating this to remove plate 4 since we don't use it for this manuscript (beyond scope).

File: 0.data_analysis/all_plates_characteristics/construct_correlation_data/concatenated_all_plates_correlations.parquet
Comment (jenna-tomkinson) [inline 0.data_analysis/all_plates_characteristics/construct_correlation_data/concatenated_all_plates_correlations.parquet:1]: Can you confirm before merging that this will only include plates 5, 3, and 3 prime, since we don't include plate 4 in manuscript.

File: 0.data_analysis/all_plates_characteristics/nbconverted/per_plate_correlations.py
Comment (jenna-tomkinson) [inline 0.data_analysis/all_plates_characteristics/nbconverted/per_plate_correlations.py:1]: It looks like this file is still using plate 4 to perform the correlations. I would recommend rerunning this notebook to remove this plate 4 info since we are using the outputted correlations for a main figure.

But if you think it is best to leave it, that is fine. I can not include it when generating the figure.

File: 1.train_models/README.md
Comment (jenna-tomkinson) [inline 1.train_models/README.md]:
```suggestion
The training and testing sets were manually chosen to create a 90-10 percent train-test split between wells while approximately retaining a uniform genotype class distribution of cells within each data split.
```

Comment (jenna-tomkinson) [reply 1.train_models/README.md]: Is this accurate to use manual here? If not, recommend clarifying what was manually done.

File: 1.train_models/README.md
Comment (jenna-tomkinson) [inline 1.train_models/README.md]:
```suggestion
We optimize the chosen model by performing a random search with stratified k-fold cross-validation using the class-balanced training set.
```

File: README.md
Comment (jenna-tomkinson) [reply README.md]: I believe it should be. We do not include any other "figures" in this module, so it makes sense to make it clear what the purpose of the module is for.

**Review by jenna-tomkinson — APPROVED** [review]
I am approving this PR but before merging, I would highly recommend making sure the correlation data includes the data that we are using in the manuscript (e.g., removing plate 4 controls).

I made a few other comments but overall this repo is much cleaner and concise, so nice job! :tada:

File: 0.data_analysis/README.md
Comment (MattsonCam) [reply 0.data_analysis/README.md]: That works, I'll make the change

File: README.md
Comment (MattsonCam) [reply README.md]: Sounds good, I guess we can provide a link to the manuscript later here

File: 0.data_analysis/README.md
Comment (MattsonCam) [reply 0.data_analysis/README.md]: I think think plate 4 is used in that analysis. Do you think we need to update this analysis? If so, should we update it in the pr, or a later pr?

File: 0.data_analysis/all_plates_characteristics/construct_correlation_data/concatenated_all_plates_correlations.parquet
Comment (MattsonCam) [reply 0.data_analysis/all_plates_characteristics/construct_correlation_data/concatenated_all_plates_correlations.parquet:1]: See above comment

File: 0.data_analysis/README.md
Comment (MattsonCam) [reply 0.data_analysis/README.md]: Read your previous comment later on (below), so I am resolving this conversation

File: 0.data_analysis/all_plates_characteristics/nbconverted/per_plate_correlations.py
Comment (MattsonCam) [reply 0.data_analysis/all_plates_characteristics/nbconverted/per_plate_correlations.py:1]: I vaguely remember deciding not to include a figure from this analysis in the manuscript. Although, I will plan on updating this analysis

File: 1.train_models/README.md
Comment (MattsonCam) [reply 1.train_models/README.md]: I think so, this is opposed to a more purely algorithmic approach. I just thought of a better way to phrase this, which will be applied shortly

File: 0.data_analysis/all_plates_characteristics/nbconverted/per_plate_correlations.py
Comment (MattsonCam) [reply 0.data_analysis/all_plates_characteristics/nbconverted/per_plate_correlations.py:1]: I think I will update this analysis in a future pr

**Comment by MattsonCam** [comment]
Thanks @jenna-tomkinson, merging now!

**Pull request merged**
